lib/Staticman.js

```diff
@@ -17,606 +15,607 @@ const Transforms = require('./Transforms')
 const uuidv1 = require('uuid/v1')
 const yaml = require('js-yaml')
 
-const Staticman = function (parameters) {
-  this.parameters = parameters
-
-  const {
-    branch,
-    repository,
-    service,
-    username,
-    version
-  } = parameters
-
-  // Initialise the Git service API
-  this.git = gitFactory.create(service, {
-    branch,
-    repository,
-    username,
-    version
-  })  
-
-  // Generate unique id
-  this.uid = uuidv1()
-
-  this.rsa = new NodeRSA()
-  this.rsa.importKey(config.get('rsaPrivateKey'))
-}
+class Staticman {
+  constructor (parameters) {
+    return (async () => {
```

Comment:
Why do we have an asynchronous constructor now?

Reply:
The Staticman constructor has to be async because it instantiates a git service which is an async operation now in the case of authenticating as a GitHub app. This is necessary since GitHub app auth requires making API calls to fetch the access token for authenticating any other requests. I'm open to other approaches if you have any suggestions.

Reply:
No, that makes sense. I just wanted to confirm that was the case. Thanks!